enzyme/Enzyme/TraceInterface.cpp

```diff
@@ -131,31 +145,75 @@ StaticTraceInterface::StaticTraceInterface(Module *M)
     }
   }
 
-  assert(newTraceFunction != nullptr && freeTraceFunction != nullptr &&
-         getTraceFunction != nullptr && getChoiceFunction != nullptr &&
-         insertCallFunction != nullptr && insertChoiceFunction != nullptr &&
-         hasCallFunction != nullptr && hasChoiceFunction != nullptr &&
-         sampleFunction != nullptr);
+  newTraceFunction->addFnAttr("enzyme_math", "enzyme_prob_if");
```

Comment:
This might be better done as "enzyme_prob_newtrace" / "enzyme_prob_freetrace" / ....